testsuite/smoke-test/src/rosetta.rs

```diff
@@ -966,6 +966,22 @@ async fn test_block() {
     .await
     .expect("Set voter should work!");
 
+    // Unlock stake
+    unlock_stake_and_wait(
```

Comment:
You should set this to the final tx so it will check the block until this tx has been saw on chain.